File: LeetCode_Marathons/LCM4/Easy/Ugly_Number/a.java
class Solution {
    private int getNWithoutDivizor(int divizor, int n) {
        while (n % divizor == 0) {
            n /= divizor;
        }
        return n;
    }

    public boolean isUgly(int n) {
        if (n <= 0) return false;
        n = getNWithoutDivizor(2, n);
        n = getNWithoutDivizor(3, n);
        n = getNWithoutDivizor(5, n);
        return n == 1;
    }
}

File: LeetCode_Marathons/LCM4/Easy/Ugly_Number/a.py
class Solution:
    def getNWithoutDivizor(self, divizor: int, n: int) -> int:
        while n % divizor == 0: n /= divizor
        return n

    def isUgly(self, n: int) -> bool:
        if n <= 0: return False
        n = self.getNWithoutDivizor(2, n)
        n = self.getNWithoutDivizor(3, n)
        n = self.getNWithoutDivizor(5, n)
        return n == 1